core/domain/question_services.py

```diff
@@ -141,14 +141,14 @@ def create_new_question_skill_link(
 
     question_skill_link_model = question_models.QuestionSkillLinkModel.create(
```

Comment:
Just to check -- in what circumstances would we need to do this outside the "if" statement below? I.e. will there be a case where the skill ID is in question.linked_skill_ids but we still need to create this model?

Reply:
Yes so the controllers first create the question (with the linked skill id) and then add the question skill link. That would have to happen outside of the if.

Reply:
But if the skill ID is already linked to the question, wouldn't there already be a question skill link model in that case?

Reply:
Oh wait, I see, in your scenario that doesn't happen.

But then, I guess ... why is there a conditional at all? Shouldn't it be more of an assertion/expectation that the skill ID is present in the linked skill IDs?

OK to keep it as is if this is just being careful, just wanted to check that this wasn't an oversight.

Reply:
I am keeping it more for care. If suppose we want to edit a question and add another skill to id then the if would pass.